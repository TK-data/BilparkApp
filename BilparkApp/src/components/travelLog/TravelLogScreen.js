import React from 'react';
import { connect } from 'react-redux';
import { StyleSheet } from 'react-native';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import { registerUserModalVisible } from '../../actions/registerUser';
import TravelLogInput from './TravelLogInput';

const styles = StyleSheet.create({
  keyboard: {
    backgroundColor: '#002776',
  },
  logo: {
    height: 25,
    width: 112,
  },
});

class TravelLogScreen extends React.Component {

  componentDidMount() {
    this.props.visibleModal(false);
  }

  render() {
    return (
      <KeyboardAwareScrollView
        style={{ backgroundColor: '#002776' }}
        resetScrollToCoords={{ x: 0, y: 0 }}
        contentContainerStyle={styles.keyboard}
      >
        <TravelLogInput />
      </KeyboardAwareScrollView>
    );
  }
}

const mapStateToProps = () => {
  return {
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    visibleModal: bool => dispatch(registerUserModalVisible(bool)),
  };
};

TravelLogScreen.navigationOptions = {
};

export default connect(mapStateToProps, mapDispatchToProps)(TravelLogScreen);
