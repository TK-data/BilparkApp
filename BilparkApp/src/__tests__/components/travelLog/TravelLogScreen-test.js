import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import TravelLogScreen from '../../../components/travelLog/TravelLogScreen';

const initialState = {
  travelLog: {
    addressFrom: '',
    addressTo: '',
    cargoValue: {
      Cargo: false,
      Comment: '',
    },
    datepickerDate: '15.5.2018',
    datepickerVisible: false,
    distance: '0km',
    formType: true,
    formValue: {
      Passenger: '0',
      Passenger1: '',
      Passenger2: '',
      Passenger3: '',
      Passenger4: '',
      Passenger5: '',
    },
    positionFrom: '0',
    positionTo: '0',
  },
  hasErrored: false,
  isLoading: true,
  auth: {
    hasErrored: false,
    isLoading: false,
    user: 'user',
  },
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing TravelLogScreen', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <TravelLogScreen />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
