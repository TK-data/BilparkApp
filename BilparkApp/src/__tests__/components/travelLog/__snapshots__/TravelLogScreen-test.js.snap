// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TravelLogScreen Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TravelLogScreen
    visibleModal={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Connect(TravelLogInput) />,
      "contentContainerStyle": Object {
        "backgroundColor": "#002776",
      },
      "enableAutomaticScroll": true,
      "enableResetScrollToCoords": true,
      "extraHeight": 75,
      "extraScrollHeight": 0,
      "keyboardOpeningTime": 250,
      "resetScrollToCoords": Object {
        "x": 0,
        "y": 0,
      },
      "style": Object {
        "backgroundColor": "#002776",
      },
      "viewIsInsideTabBar": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Connect(TravelLogInput) />,
        "contentContainerStyle": Object {
          "backgroundColor": "#002776",
        },
        "enableAutomaticScroll": true,
        "enableResetScrollToCoords": true,
        "extraHeight": 75,
        "extraScrollHeight": 0,
        "keyboardOpeningTime": 250,
        "resetScrollToCoords": Object {
          "x": 0,
          "y": 0,
        },
        "style": Object {
          "backgroundColor": "#002776",
        },
        "viewIsInsideTabBar": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
